hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1118,9 +1118,9 @@ public StoreFileWriter createWriterInTmp(long maxKeyCount, Compression.Algorithm
       boolean shouldDropBehind) throws IOException {
     final CacheConfig writerCacheConf;
     if (isCompaction) {
-      // Don't cache data on write on compactions.
+      // Don't cache data on write on compactions, unless specifically configured to do so
       writerCacheConf = new CacheConfig(cacheConf);
-      writerCacheConf.setCacheDataOnWrite(false);
+      writerCacheConf.setCacheDataOnWrite(cacheConf.shouldCacheCompactedBlocksOnWrite());
```

Comment:
@ramkrish86 Do you thing there is a need to enable cacheIndexesOnWrite and cacheBloomsOnWrite also. may be we can declare a method like enableCacheOnWrite in CacheConfig to enable them all.

Reply:
Good point. Currently if cacheDAtaOnWrite is true, still the cacheBloomOnWrite and cacheIndexOnWrite can be false. So may be if cacheDataOnWrite is true we should make the other two also as true? And this to happen even durng flushes and not only compactions? Just discussed with @anoopsjohn  offline and if needed we can do this in another JIRA. @chenxu14  - your thoughts?

Reply:
OK, looking forward to